Sources/core/objects/Movable.h

```diff
@@ -34,7 +34,7 @@ class IMovable : public IOnMapObject
     void processMove();//for move
     virtual bool TryMove(Dir direct);
     bool CheckMoveTime();
-    bool CheckPassable();
+    virtual bool CheckPassable();
     bool MainMove();
```

Comment:
It is not a really big deal to make a function virtual. Now it will be overrided only in projectiles, but it may be helpful later for some other needs.
This new function will be called (as far as I can see) only from `Projectile::Process`, so you can keep `Delete` in `Projectile::Process` and it will be no differences with `ProjectileStatusCheck` in game sense. If you have implied that the new function will be global (in some big base class like `IMovable` or `IOnMapObject`) then it is much worse than making some existing function virtual.